functorch/functorch/_src/aot_autograd.py

```diff
@@ -174,146 +175,192 @@ def call_func_with_args(f, args, steal_args=False):
-                            # Functionalize the foward backward graph. First create a
-                            # fake fn to make functionalize happy
-                            def fake_fn(primals, tangents):
-                                return fx_g(primals, tangents)
+    if config.debug_joint:
+        print(fx_g.code)
 
-                            fx_g = make_fx(functionalize(fake_fn))(*joint_inputs)
+    with track_graph_compiling("joint"):
+        fw_module, bw_module = aot_config.partition_fn(fx_g, joint_inputs)
 
-                if config.debug_joint:
-                    print(fx_g.code)
+    if config.debug_graphs:
+        print(fw_module.code, bw_module.code)
 
-                with track_graph_compiling("joint"):
-                    fw_module, bw_module = partition_fn(fx_g, joint_inputs)
+    with track_graph_compiling("forward"):
+        compiled_fw = aot_config.fw_compiler(fw_module, flat_args)
 
-                if config.debug_graphs:
-                    print(fw_module.code, bw_module.code)
+    # TODO: Delay this backwards compilation until the backwards pass
+    with torch.no_grad():
+        fw_outs = call_func_with_args(compiled_fw, flat_args)
 
-                with track_graph_compiling("forward"):
-                    compiled_fw = fw_compiler(fw_module, flat_tensor_args)
+    if config.debug_partitioner:
+        activation_sizes = 0
+        for out in fw_outs[num_outs:]:
+            if isinstance(out, torch.Tensor):
+                activation_sizes += out.storage().nbytes()
+        print(f"Real Activations Stored(GB): {activation_sizes/1e9}")
 
-                fw_outs = call_func_with_args(compiled_fw, flat_tensor_args)
-                if config.debug_partitioner:
-                    activation_sizes = 0
-                    for out in fw_outs[num_outs:]:
-                        if isinstance(out, torch.Tensor):
-                            activation_sizes += out.storage().nbytes()
-                    print(f"Real Activations Stored(GB): {activation_sizes/1e9}")
+    bw_args = fw_outs[num_outs:] + fw_outs[0:num_outs]
+    with track_graph_compiling("backward", True):
+        compiled_bw = aot_config.bw_compiler(bw_module, bw_args)
+    torch._C._jit_set_autocast_mode(old_jit_autocast_flag)
 
-                bw_args = fw_outs[num_outs:] + fw_outs[0:num_outs]
-                with track_graph_compiling("backward", True):
-                    compiled_bw = bw_compiler(bw_module, bw_args)
-            else:
-                fw_outs = call_func_with_args(compiled_fw, flat_tensor_args)
-            torch._C._jit_set_autocast_mode(old_jit_autocast_flag)
+    class CompiledFunction(torch.autograd.Function):
+        @staticmethod
+        @disable_torchdynamo
+        def forward(ctx, *flat_tensor_args):
+            # Disable the JIT Autocast flag to prevent re-autocasting of jitted graph.
+            # TODO - Remove when https://github.com/pytorch/functorch/pull/794 is fixed.
```

Comment:
First line of comment doesn't seem relevant anymore? autocast flag is being disabled inside the function